pkg/api/cachedcsi.go

```diff
@@ -0,0 +1,203 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"path"
+	"path/filepath"
+	"syscall"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/gadget-inc/dateilager/internal/key"
+	"github.com/gadget-inc/dateilager/internal/logger"
+	"github.com/gadget-inc/dateilager/pkg/version"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	DriverName = "com.gadget.dateilager.cached"
+)
+
+// GetPluginInfo returns metadata of the plugin
+func (c *Cached) GetPluginInfo(ctx context.Context, req *csi.GetPluginInfoRequest) (*csi.GetPluginInfoResponse, error) {
+	resp := &csi.GetPluginInfoResponse{
+		Name:          DriverName,
+		VendorVersion: version.Version,
+	}
+
+	return resp, nil
+}
+
+// GetPluginCapabilities returns available capabilities of the plugin
+func (c *Cached) GetPluginCapabilities(ctx context.Context, req *csi.GetPluginCapabilitiesRequest) (*csi.GetPluginCapabilitiesResponse, error) {
+	resp := &csi.GetPluginCapabilitiesResponse{
+		Capabilities: []*csi.PluginCapability{},
```

Comment:
Sounds like this will support `CONTROLLER_SERVICE` and `CREATE_DELETE_VOLUME`?

Reply:
I don't think the CONTROLLER_SERVICE -- just the node service.